### PR TITLE
#12

### PR DESCRIPTION
**Pull request by timon-schelling**

### CODE REVIEW

**Comment by 616slayer616** [comment]
Looks good so far.
Could you add an example for configuring multiple minification tasks in the Readme?

**Comment by timon-schelling** [comment]
> Looks good so far.
> Could you add an example for configuring multiple minification tasks in the Readme?

Ok, in this branche or in rerelease?

**Comment by timon-schelling** [comment]
should the package name be changed?

**Comment by 616slayer616** [comment]
> > Looks good so far.
> > Could you add an example for configuring multiple minification tasks in the Readme?
> 
> Ok, in this branche or in rerelease?

I would do it in this one.

**Comment by 616slayer616** [comment]
> should the package name be changed?

Oh yes. forgot that.

I propose:
`src/main/kotlin/org/gradlewebtools/minify/`

**Comment by timon-schelling** [comment]
Why does the sonarqube task throws this error?
```
./gradlew sonarqube
...
> Task :sonarqube FAILED
* What went wrong:
Execution failed for task ':sonarqube'.
> Could not find a default branch to fall back on.
```

**Comment by 616slayer616** [comment]
I think it's because there is no develop branch for this project yet.
I set up a new organization and sonar project. But it needs a develop run first.

**Comment by timon-schelling** [comment]
What needs to be done before release?
Can I help with something else?

File: src/main/kotlin/org/padler/gradle/minify/minifier/JsMinifier.kt [hunk 1]
```diff
@@ -21,7 +21,10 @@ class JsMinifier(override var minifierOptions: JSMinifierOptions = JSMinifierOpt
         if (minifierOptions.createSourceMaps) {
             sourcemapFile = File(dstFile.absolutePath + ".map")
             options.setSourceMapOutputPath(sourcemapFile.absolutePath)
-            options.setSourceMapLocationMappings(listOf(RelativePathLocationMapping()))
+            options.setSourceMapLocationMappings(listOf(SourceMap.LocationMapping { location ->
+                val index = location.lastIndexOf('/')
```
Comment (timon-schelling) [inline src/main/kotlin/org/padler/gradle/minify/minifier/JsMinifier.kt]: could be replaces with substringAfterLast("/").takeIf{ notEmpty() }

File: src/main/kotlin/org/gradlewebtools/minify/minifier/js/JsMinifier.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    override fun minifyFile(srcFile: File, dstFile: File) {
+        val compiler = Compiler()
+        setOptions()
+        val externals = AbstractCommandLineRunner.getBuiltinExterns(CompilerOptions().environment)
+        val sourceFile = SourceFile.fromFile(srcFile.absolutePath)
+        val sourcemapFile = File(dstFile.absolutePath + ".map")
+        if (minifierOptions.createSourceMaps) {
+            options.setSourceMapOutputPath(sourcemapFile.absolutePath)
+            options.setSourceMapLocationMappings(listOf(SourceMap.LocationMapping {
+                val index = it.lastIndexOf('/')
```
Comment (timon-schelling) [inline src/main/kotlin/org/gradlewebtools/minify/minifier/js/JsMinifier.kt]: could be replaces with substringAfterLast("/").takeIf{ notEmpty() }

**Review by 616slayer616 — APPROVED** [review]

**Comment by 616slayer616** [comment]
The Readme looks good!

I think you can merge this branch.
I will then update my "Re Release" branch and check if everything looks alright.

Then we could release it

**Pull request merged**